csv-validator-ui/src/main/scala/uk/gov/nationalarchives/csv/validator/ui/CsvValidatorUi.scala

```diff
@@ -229,7 +232,39 @@ object CsvValidatorUi extends SimpleSwingApplication {
     private val layout = new DesignGridLayout(peer)
 
     private val lblCsvFile = new Label("CSV file:")
-    private val txtCsvFile = new TextField(30)
+    private val txtCsvFile = new JTextField(30)
+
+    final class FileDropHandler(fileExt: String, label: Label) extends TransferHandler {
+      override def canImport(support: TransferHandler.TransferSupport): Boolean = support.getDataFlavors.exists(_.isFlavorJavaFileListType)
+
+      @SuppressWarnings(Array("unchecked"))
+      override def importData(support: TransferHandler.TransferSupport): Boolean = {
+        if (!this.canImport(support)) false
+        else {
+          val potentialFiles = Try {
+            support.getTransferable.getTransferData(DataFlavor.javaFileListFlavor).asInstanceOf[util.List[File]].asScala.toList
+          }
+          potentialFiles match {
+            case Failure(ex) => false
+            case Success(files) =>
+              lazy val pathOfFirstFile = files.head.getAbsolutePath
+              if(files.length != 1) {
+                outputToReport(s"Error: Please drag only 1 file into the '${label.text}' text box.")
+                false
+              }
+              else if(!pathOfFirstFile.endsWith(fileExt)) {
+                outputToReport(s"Error: Please drag only '$fileExt' files into the '${label.text}' text box.")
+                false
+              }
+              else {
+                (if(fileExt == ".csv") txtCsvFile else txtCsvSchemaFile).setText(pathOfFirstFile)
+                true
+              }
+          }
+        }
+      }
+    }
+    txtCsvFile.setTransferHandler(new FileDropHandler(".csv", lblCsvFile))
```

Comment:
Can you please move this line above the `FileDropDownHandler` class definition, just so that it stays close to the textBox creation from readability perspective